### PR TITLE
Rename apply

### PR DESCRIPTION
**Pull request by stew**
rename the "apply" method on the Apply typeclass to "ap". apply is such a special name in the scala world that this method, which is rarely useful to users, should probably not use this name.

### CODE REVIEW

**Comment by fthomas** [comment]
:+1:

File: core/src/main/scala/cats/Apply.scala [hunk 1]
```diff
@@ -14,13 +14,13 @@ trait Apply[F[_]] extends Functor[F] with ApplyArityFunctions[F] { self =>
    * Given a value and a function in the Apply context, applies the
    * function to the value.
    */
-  def apply[A, B](fa: F[A])(f: F[A => B]): F[B]
+  def ap[A, B](fa: F[A])(f: F[A => B]): F[B]
 
   /**
-   * apply2 is a binary version of apply, defined in terms of apply.
+   * apply2 is a binary version of ap, defined in terms of ap.
```
Comment (ceedubs) [inline core/src/main/scala/cats/Apply.scala]: Should be "ap2 is a binary..."

**Comment by ceedubs** [comment]
Looks good to merge to me after the one missing apply2 => ap2 conversion I pointed out is changed.

**Comment by ceedubs** [comment]
:+1:

**Pull request merged**